domain/src/main/java/org/oppia/android/domain/hintsandsolution/HintHandler.kt

```diff
@@ -0,0 +1,210 @@
+package org.oppia.android.domain.hintsandsolution
```

Comment:
Just want to clarify -- this is an exact copy of the existing hint handler correct?

Reply:
No this is not an exact copy of the previous hint handler. 

The previous hint handler was responsible for the following
- compute which hint (if any) should be revealed to the user.
- schedule a new timer, observe it and show new hints when the timer runs out.

The new hint handler computes which hint should be revealed but it is no longer responsible to schedule a task, instead the hint handler computes the delay after which the hint should be revealed. The timer is scheduled by the exploration progress controller.